src/core/server/server.h

```diff
@@ -131,6 +236,48 @@ class Server : public ServerInterface,
     /// Sets a closure to be invoked by the listener when its destruction
     /// is complete.
     virtual void SetOnDestroyDone(grpc_closure* on_destroy_done) = 0;
+
+    Server* server() const { return server_; }
+
+   protected:
+    // Adds a LogicalConnection to the listener and updates the channel args if
+    // needed.
+    void AddLogicalConnection(
+        absl::AnyInvocable<
```

Comment:
It seems a little ugly to pass in a callback here to create the connection.  Is there a reasonable way to restructure this such that the transport creates the connection and passes it directly?

If we keep this, I think it can be `absl::FunctionRef<>` instead of `absl::AnyInvocable<>`, because it's always called synchronously.